src/ngraph/runtime/gpu/gpu_tensor.cpp

```diff
@@ -39,11 +39,15 @@ runtime::gpu::GPUTensor::GPUTensor(const ngraph::element::Type& element_type,
         std::make_shared<ngraph::descriptor::layout::DenseTensorLayout>(*m_descriptor));
 
     m_buffer_size = shape_size(shape) * element_type.size();
-    if (memory_pointer != nullptr)
+    if (memory_pointer != nullptr && is_device_pointer(memory_pointer))
```

Comment:
Have one if with memory_pointer != nullptr then inside check for is_device_pointer. It will look a little cleaner.

Reply:
agree. fixed.